### PR TITLE
After adding the Open Telemetry agent the plugin crashes https://gith…

### PR DESCRIPTION
**Pull request by EdwardKuenen**
…ub.com/rh-messaging/artemis-prometheus-metrics-plugin/issues/12

### CODE REVIEW

File: artemis-prometheus-metrics-plugin-servlet/pom.xml [hunk 1]
```diff
@@ -24,7 +24,7 @@ under the License.
    <parent>
       <groupId>com.redhat.amq-broker</groupId>
       <artifactId>artemis-prometheus-metrics-pom</artifactId>
-      <version>2.0.0</version>
+      <version>2.0.1</version>
```
Comment (brusdev) [inline artemis-prometheus-metrics-plugin-servlet/pom.xml:27]: Could you move the version change to separate commit, i.e. https://github.com/rh-messaging/artemis-prometheus-metrics-plugin/commit/d0a9b3ea0e911d2cbeb89fb64d0b27bc240b55ea

File: artemis-prometheus-metrics-plugin-servlet/pom.xml [hunk 2]
```diff
@@ -35,19 +35,14 @@ under the License.
       <dependency>
          <groupId>org.apache.tomcat</groupId>
          <artifactId>tomcat-servlet-api</artifactId>
-         <version>8.5.5</version>
```
Comment (brusdev) [inline artemis-prometheus-metrics-plugin-servlet/pom.xml:38]: Could you move the maven impovement to a separate commit?

File: artemis-prometheus-metrics-plugin/src/main/java/com/redhat/amq/broker/core/server/metrics/plugins/ArtemisPrometheusMetricsPlugin.java [hunk 3]
```diff
@@ -26,11 +26,11 @@
 
 public class ArtemisPrometheusMetricsPlugin implements ActiveMQMetricsPlugin {
 
-   MeterRegistry meterRegistry;
+   private MeterRegistry meterRegistry;
 
    @Override
-   public ActiveMQMetricsPlugin init(Map<String, String> options) {
-      this.meterRegistry = new PrometheusMeterRegistry(PrometheusConfig.DEFAULT);
+   public ActiveMQMetricsPlugin init(final Map<String, String> options) {
```
Comment (brusdev) [inline artemis-prometheus-metrics-plugin/src/main/java/com/redhat/amq/broker/core/server/metrics/plugins/ArtemisPrometheusMetricsPlugin.java:32]: Is this change required?

**Review by brusdev — COMMENTED** [review]
@EdwardKuenen this is more than a simple fix, nice work!!!

File: artemis-prometheus-metrics-plugin/src/main/java/com/redhat/amq/broker/core/server/metrics/plugins/ArtemisPrometheusMetricsPlugin.java (hunk 3 above)
Comment (EdwardKuenen) [reply artemis-prometheus-metrics-plugin/src/main/java/com/redhat/amq/broker/core/server/metrics/plugins/ArtemisPrometheusMetricsPlugin.java:32]: No, not required, but making a variable final if it should not change is my preference. Should I revert it?

Comment (brusdev) [reply artemis-prometheus-metrics-plugin/src/main/java/com/redhat/amq/broker/core/server/metrics/plugins/ArtemisPrometheusMetricsPlugin.java:32]: I see your point but I usually avoid changes not required or not related to the commit subject to avoid confusion for future contributors that need to review the commit

File: artemis-prometheus-metrics-plugin-servlet/pom.xml (hunk 2 above)
Comment (brusdev) [reply artemis-prometheus-metrics-plugin-servlet/pom.xml:38]: Separate PR created: https://github.com/rh-messaging/artemis-prometheus-metrics-plugin/pull/15

**Comment by brusdev** [comment]
@EdwardKuenen  I see you created new PRs to split the commits:
 https://github.com/rh-messaging/artemis-prometheus-metrics-plugin/pull/14
 https://github.com/rh-messaging/artemis-prometheus-metrics-plugin/pull/15
 
 Feel free to close this PR if it is no required anymore

File: artemis-prometheus-metrics-plugin-servlet/pom.xml (hunk 1 above)
Comment (brusdev) [reply artemis-prometheus-metrics-plugin-servlet/pom.xml:27]: I'll push a commit to bump the version when your fix will be merged.
